vassal-app/src/main/java/VASSAL/build/module/GameRefresher.java

```diff
@@ -93,24 +93,38 @@ public final class GameRefresher implements CommandEncoder, GameComponent {
   public static final String COMMAND_PREFIX = "DECKREPOS" + DELIMITER; //$NON-NLS-1$
 
   public static final String USE_NAME = "UseName";
+  public static final String FIX_GPID = "fixGPID";
   public static final String USE_LABELER_NAME = "UseLabelerName";
   public static final String USE_LAYER_NAME = "UseLayerName";
   public static final String USE_ROTATE_NAME = "UseRotateName";
   public static final String TEST_MODE = "TestMode";
   public static final String DELETE_NO_MAP = "DeleteNoMap";
   public static final String REFRESH_DECKS = "RefreshDecks";
+  public static final String REFRESH_PIECES = "RefreshPieces";
   public static final String DELETE_OLD_DECKS = "DeleteOldDecks";
   public static final String ADD_NEW_DECKS = "AddNewDecks";
+  public static final String USE_HOTKEY = "UseHotkey";
+  public static final String SUPPRESS_INFO_REPORTS = "SuppressInfoReports";
 
   private Action refreshAction;
   private final GpIdSupport gpIdSupport;
   private GpIdChecker gpIdChecker;
   private RefreshDialog dialog;
   private int updatedCount;
-  private int notFoundCount;
-  private int noStackCount;
-  private int noMapCount;
 
+  private int totalCount;
+  private int totalDecks;
+
+  int notFoundCount; // shared to PDS refresher
+  int noGpIdMatch; // shared to PDS refresher
+  int noStackCount; // shared to PDS refresher - not used!!!
+  int noMapCount; // shared to PDS refresher - not used!!!
+  int notOwnedCount; // shared to PDS refresher
+  int notVisibleCount; // shared to PDS refresher
+  int deckWarnings; // shared to PDS refresher
```

Comment:
What's the deal with these? They need a visibility modifier. Should they be private? protected?

Reply:
These variables and the 3 immediately preceding them are various types of error accumulated during a refresh run. Visibility must at least allow PreDefinedSetup.java to access them - it uses them to return an error count to RefreshPreDefinedSetupDialog. This is all so that the user gets a summary at the end rather than having to trawl through all the refresh logs.

Private doesn't work so I guess Protected ? IntelliJ doesn't approve of that and warns "Class member declared 'protected' in 'final' class"  hence I accepted it's suggestion, which removed the Protected modifier:

<img width="433" alt="image" src="https://github.com/vassalengine/vassal/assets/58135975/3edda4da-6c29-45d6-8386-31ffd2301a80">

i can added the Protected modifiers back in  unless there's some other way I should be doing this.

Reply:
All see that you're doing with these variables outside of `GameRefresher` is summing them in one location. If that's all you need, then they should be made private and you can create a public method inside of which you compute the sum, and then call that method from outside of `GameRefresher`.

Reply:
Thanks. I've pushed the fix to this.

Whilst doing the fix, I've noticed that somehow in a commit on 7th Dec, I've managed to over-write the change history in GameRefresher. I presume it's happened whilst I was working around a glitch in IntelliJ.  I think I may need to unwind or re-do this PR to fix this.

<img width="481" alt="image" src="https://github.com/vassalengine/vassal/assets/58135975/dfa6c0e8-30c8-467a-919c-cac7e049957a">